### PR TITLE
Core: Switch tests to Junit5 in deletes,expressions,auth packages

### PR DESCRIPTION
**Pull request by skytin1004**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I switched all the files in the deletes package from JUnit4 to JUnit5.

### CODE REVIEW

**Comment by nastra** [comment]
@skytin1004 I think it's ok to combine multiple packages as long as the PR doesn't get too big

**Comment by skytin1004** [comment]
@nastra Thanks for your review. Based on your review of the other PR I posted, I modified the files and combined the files into single PR.

File: core/src/test/java/org/apache/iceberg/deletes/TestEqualityFilter.java [hunk 1]
```diff
@@ -63,14 +64,17 @@ public void testEqualitySetFilterLongColumn() {
...
         Lists.newArrayList(
             Deletes.filter(
                 ROWS,
                 row -> Row.of(row.get(0, Long.class)),
-                Deletes.toEqualitySet(deletes, ROW_SCHEMA.select("id").asStruct()))));
+                Deletes.toEqualitySet(deletes, ROW_SCHEMA.select("id").asStruct())));
+
+    assertThat(actual)
+        .as("Filter should produce expected rows")
+        .usingRecursiveComparison()
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/deletes/TestEqualityFilter.java:76]: why do we need this one? The test passes for me without having `usingRecursiveComparison()`

**Review by nastra — COMMENTED** [review]
mostly LGTM, I think we don't need to specify `usingRecursiveComparison()` in `TestEqualityFilter`

**Pull request closed without merging**